gcloud/vision/image.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2016 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from base64 import b64encode
+
+
+class Image(object):
+    """Image representation containing either the source or content of the
+    image that will get annotated.
+
+    :type image_source: str
+    :param image_source: A string which can be a URL, a Google Cloud Storage
+                         path, or a byte stream/string of the image.
+
+    See:
+    https://cloud.google.com/vision/reference/rest/v1/images/annotate#Image
+    """
+
+    def __init__(self, image_source, client):
+        """Initialize Image class
+
+        :type image_source: str
+        :param image_source: String or Bytes
```

Comment:
I think this constructor needs to have optional params for the different cases:

- `content` would be raw bytes
- `gs_uri` would be either a `gs:/bucket/blob`-style URI

I think for now we should punt on filenames (the user can type: `with open(filename) as stream: image = Image(content=stream.read())`, and *definitely* punt on non-GCS URLs (we don't want to be in the business of fetching them on the user's behalf).